3518 Resurrect the IoT tutorial (#3543)
timescale/docs: tutorials/simulate-iot-sensor-data.md

---

> title: How to simulate a basic IoT sensor dataset

Reviewer: ```suggestion
title: Simulate an IoT sensor dataset
```

> excerpt: Test your database by simulating a dataset

Reviewer: ```suggestion
excerpt: Simulate an IOT dataset in your Timescale Cloud service
```

products: [cloud, mst, self_hosted]
keywords: [IoT, simulate]
---

import ImportPrerequisites from "versionContent/_partials/_migrate_import_prerequisites.mdx";

# Simulate an IoT sensor dataset

> The Internet of Things (IoT) describes a trend where computing capabilities are embedded into physical objects (IoT devices), ranging from light bulbs to oil wells. Many of those IoT devices collect sensor data about their environment and generate time-series datasets with relational metadata.

Reviewer: ```suggestion
The Internet of Things (IoT) describes a trend where computing capabilities are embedded into IoT devices. That is, physical objects, ranging from light bulbs to oil wells. Many IoT devices collect sensor data about their environment and generate time-series datasets with relational metadata.
```

> It is often necessary to simulate such datasets, for example, when

Reviewer: ```suggestion
It is often necessary to simulate IoT datasets. For example, when you are 
```

> testing a new system. This tutorial shows how to simulate a basic dataset with $CLOUD_LONG and then run simple queries on it. 

Reviewer: ```suggestion
testing a new system. This tutorial shows how to simulate a basic dataset in your $SERVICE_LONG, and then run simple queries on it. 
```

To simulate a more advanced dataset, see [Time-series Benchmarking Suite (TSBS)][tsbs].

## Prerequisites

> Before starting this procedure:

Reviewer: ```suggestion
To follow this tutorial, you need to:
```

- [Create a target Timescale Cloud service][create-a-service].
- [Connect to your service][connect-to-service].

## Simulate a dataset

<Procedure>

> To simulate a dataset, run the following queries. You can do so from the data mode, the ops mode SQL editor, or using `psql`:

Reviewer: ```suggestion
To simulate a dataset, run the following queries:
```
I don't think we need the list of editors as it is in the Prerequisites

     last(temperature, time) AS last_temp,
     AVG(cpu) AS avg_cpu
   FROM sensor_data JOIN sensors on sensor_data.sensor_id = sensors.id
   GROUP BY period, sensors.location;
   ```
   
   Sample output:
   
   ```
    location |         period         |     avg_temp     |     last_temp     |      avg_cpu      
   ----------+------------------------+------------------+-------------------+-------------------
    ceiling  | 20120-03-31 15:30:00+00 | 25.4546818090603 |  24.3201029952615 | 0.435734559316188
    floor    | 2020-03-31 15:30:00+00 | 43.4297036845237 |  79.9925176426768 |  0.56992522883229
    ceiling  | 2020-03-31 16:00:00+00 | 53.8454438598516 |  43.5192013625056 | 0.490728285357666
    floor    | 2020-03-31 16:00:00+00 | 47.0046211887772 |  23.0230117216706 |  0.53142289724201
    ceiling  | 2020-03-31 16:30:00+00 | 58.7817596504465 |  63.6621567420661 | 0.488188337767497
    floor    | 2020-03-31 16:30:00+00 |  44.611586847653 |  2.21919436007738 | 0.434762630766879
    ceiling  | 2020-03-31 17:00:00+00 | 35.7026890735142 |  42.9420990403742 | 0.550129583687522
    floor    | 2020-03-31 17:00:00+00 | 62.2794370166957 |  52.6636955793947 | 0.454323202022351
   ...

>    ```

Reviewer: And an ending sentence saying "wel done, you have done bla bla"

[create-a-service]: /getting-started/:currentVersion:/services/#create-a-timescale-cloud-service

> [connect-to-service]: /getting-started/:currentVersion:/services/#connect-to-your-service

Reviewer: ```suggestion
[connect-to-service]: /getting-started/:currentVersion:/run-queries-from-console/
```
